server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -128,42 +127,41 @@ public Collection<ExecutorBuilder> builders() {
-        for (final Map.Entry<String, ExecutorBuilder> entry : builders.entrySet()) {
-            final ExecutorHolder executorHolder = entry.getValue().build(settings);
-            if (executors.containsKey(executorHolder.info.name())) {
+
+        builders = List.of(
+            new ScalingExecutorBuilder(Names.GENERIC, 4, genericThreadPoolMax, TimeValue.timeValueSeconds(30)),
+            new FixedExecutorBuilder(settings, Names.WRITE, availableProcessors, 200),
+            new PrioFixedExecutorBuilder(settings, Names.SEARCH, searchThreadPoolSize(availableProcessors), 1000),
+            new ScalingExecutorBuilder(Names.MANAGEMENT, 1, 5, TimeValue.timeValueMinutes(5)),
+            // no queue athis means clients will need to handle rejections on listener queue even if the operation succeeded
+            // the assumpon here is that the listeners should be very lightweight on the listeners side
+            new FixedExecutorBuilder(settings, Names.LISTENER, halfProcMaxAt10, -1),
+            new ScalingExecutorBuilder(Names.FLUSH, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)),
+            new ScalingExecutorBuilder(Names.REFRESH, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5)),
+            new ScalingExecutorBuilder(Names.WARMER, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)),
+            new ScalingExecutorBuilder(Names.SNAPSHOT, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)),
+            new ScalingExecutorBuilder(Names.FETCH_SHARD_STARTED, 1, 2 * availableProcessors, TimeValue.timeValueMinutes(5)),
+            new FixedExecutorBuilder(settings, Names.FORCE_MERGE, 1, -1),
+            new ScalingExecutorBuilder(Names.FETCH_SHARD_STORE, 1, 2 * availableProcessors, TimeValue.timeValueMinutes(5)),
+            new FixedExecutorBuilder(settings, Names.LOGICAL_REPLICATION, searchThreadPoolSize(availableProcessors), 100)
+        );
+        HashMap<String, ExecutorHolder> executors = HashMap.newHashMap(builders.size() + 1);
+        for (ExecutorBuilder builder : builders) {
+            final ExecutorHolder executorHolder = builder.build(settings);
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("created thread pool: {}", builder.formatInfo(executorHolder.info));
+            }
+            ExecutorHolder previous = executors.put(executorHolder.info.name(), executorHolder);
+            if (previous != null) {
                 throw new IllegalStateException("duplicate executors with name [" + executorHolder.info.name() + "] registered");
             }
-            LOGGER.debug("created thread pool: {}", entry.getValue().formatInfo(executorHolder.info));
-            executors.put(entry.getKey(), executorHolder);
         }
-
         executors.put(Names.SAME, new ExecutorHolder(EsExecutors.directExecutor(), new Info(Names.SAME, ThreadPoolType.DIRECT)));
-        this.executors = unmodifiableMap(executors);
+        this.executors = Map.copyOf(executors);
```

Comment:
Out of curiosity, is there any benefit from this change?

Reply:
The internal representation is a bit different - Map.copyOf uses one array for both keys and values. Downside are the additional copy operations, but given that this only happens on startup it should be fine.